core/src/main/scala/design/hamu/Board.scala

```diff
@@ -0,0 +1,63 @@
+package design.hamu
+
+//   Y
+//   ^
+// 2 | _ _ _
+// 1 | _ _ _
+// 0 | _ _ _
+//   +-------> X
+//     0 1 2
+
+class Board(var vecofvec: Vector[Vector[Char]]) {
+
+  // public function. call to init a new board with a new vecofvec, with one change in the given coordinate.
+  // returns same board with no changes if out of bounds
+  def move(targetX: Int, targetY: Int, targetPiece: Char): Board = {
+    try {
+      val oldCol = vecofvec(targetY)
+      val newCol = oldCol.updated(targetX, targetPiece)
+      val newBoard = vecofvec.updated(targetY, newCol)
+      new Board(newBoard)
+    } catch {
+      case x: IndexOutOfBoundsException => this
```

Comment:
Maybe we should just let the exception be thrown if out of bounds? Since unexpected behavior should fail?